lib/core/versions/latest/CreateOperation.ts

```diff
@@ -89,24 +99,36 @@ export default class CreateOperation implements OperationModel {
    * The `operationBuffer` given is assumed to be valid and is assigned to the `operationBuffer` directly.
    * NOTE: This method is purely intended to be used as an optimization method over the `parse` method in that
    * JSON parsing is not required to be performed more than once when an operation buffer of an unknown operation type is given.
+   * @param anchorFileMode If set to true, then `operationData` and `type` properties is expected to be absent.
    */
-  public static async parseObject (operationObject: any, operationBuffer: Buffer): Promise<CreateOperation> {
-    const properties = Object.keys(operationObject);
-    if (properties.length !== 3) {
-      throw new SidetreeError(ErrorCode.CreateOperationMissingOrUnknownProperty);
+  public static async parseObject (operationObject: any, operationBuffer: Buffer, anchorFileMode: boolean): Promise<CreateOperation> {
+    let expectedPropertyCount = 3;
+    if (anchorFileMode) {
+      expectedPropertyCount = 1;
     }
 
-    if (operationObject.type !== OperationType.Create) {
-      throw new SidetreeError(ErrorCode.CreateOperationTypeIncorrect);
+    const properties = Object.keys(operationObject);
+    if (properties.length !== expectedPropertyCount) {
+      throw new SidetreeError(ErrorCode.CreateOperationMissingOrUnknownProperty);
     }
 
     const encodedSuffixData = operationObject.suffixData;
-    const encodedOperationData = operationObject.operationData;
     const suffixData = await CreateOperation.parseSuffixData(encodedSuffixData);
-    const operationData = await CreateOperation.parseOperationData(operationObject.operationData);
 
-    const didUniqueSuffix = CreateOperation.computeHash(operationObject.suffixData);
-    return new CreateOperation(operationBuffer, didUniqueSuffix, encodedSuffixData, encodedOperationData, suffixData, operationData);
+    // If not in anchor file mode, we need to validate `type` and `operationData` properties.
+    let encodedOperationData = undefined;
+    let operationData = undefined;
+    if (!anchorFileMode) {
```

Comment:
So if `anchorFileMode === true` then we will assume that operationData is valid (if present)? or is it that there wouldn't be any operationData there?

Reply:
It's the latter: operationData won't be there (in the anchor file).